src/pages/settings/Profile/CustomStatus/StatusPage.js

```diff
@@ -0,0 +1,120 @@
+import React, {useMemo, useCallback, useEffect} from 'react';
+import {View} from 'react-native';
+import {withOnyx} from 'react-native-onyx';
+import lodashGet from 'lodash/get';
+import withCurrentUserPersonalDetails, {withCurrentUserPersonalDetailsPropTypes} from '../../../../components/withCurrentUserPersonalDetails';
+import MenuItemWithTopDescription from '../../../../components/MenuItemWithTopDescription';
+import StaticHeaderPageLayout from '../../../../components/StaticHeaderPageLayout';
+import * as Expensicons from '../../../../components/Icon/Expensicons';
+import withLocalize from '../../../../components/withLocalize';
+import MenuItem from '../../../../components/MenuItem';
+import Button from '../../../../components/Button';
+import Text from '../../../../components/Text';
+import Navigation from '../../../../libs/Navigation/Navigation';
+import * as User from '../../../../libs/actions/User';
+import MobileBackgroundImage from '../../../../../assets/images/money-stack.svg';
+import themeColors from '../../../../styles/themes/default';
+import useLocalize from '../../../../hooks/useLocalize';
+import styles from '../../../../styles/styles';
+import compose from '../../../../libs/compose';
+import ONYXKEYS from '../../../../ONYXKEYS';
+import ROUTES from '../../../../ROUTES';
+
+const propTypes = {
+    ...withCurrentUserPersonalDetailsPropTypes,
+};
+
+function StatusPage({draftStatus, currentUserPersonalDetails}) {
+    const localize = useLocalize();
+    const currentUserEmojiCode = lodashGet(currentUserPersonalDetails, 'status.emojiCode', '');
+    const currentUserStatusText = lodashGet(currentUserPersonalDetails, 'status.text', '');
+    const draftEmojiCode = lodashGet(draftStatus, 'emojiCode');
+    const draftText = lodashGet(draftStatus, 'text');
+
+    const defaultEmoji = draftEmojiCode || currentUserEmojiCode;
+    const defaultText = draftText || currentUserStatusText;
+    const customStatus = defaultEmoji ? `${defaultEmoji} ${defaultText}` : '';
+    const hasDraftStatus = !!draftEmojiCode || !!draftText;
+
+    const updateStatus = useCallback(() => {
+        const endOfDay = new Date();
+        endOfDay.setHours(23, 59, 59, 999);
+        User.updateCustomStatus({text: defaultText, emojiCode: defaultEmoji, clearAfter: endOfDay.toISOString()});
+
+        User.clearDraftCustomStatus();
+        Navigation.goBack(ROUTES.SETTINGS);
+    }, [defaultText, defaultEmoji]);
+
+    const clearStatus = () => {
+        User.clearCustomStatus();
+        User.clearDraftCustomStatus();
+    };
+
+    const footerComponent = useMemo(
```

Comment:
i will say that here we do not need memoization - because you returning JSX which is not a heavy calculation - also moving this code directly in render - removes not needed useCallback for updateStatus - because Button component is still a Class which is extended on Component (not PureComponent) and it will be re-rendered in any case.

Reply:
As we gradually migrate all Class components to FC, it's prudent to prepare our code accordingly. Even if the Button component were to be a highly optimized FC, the `onPress={updateStatus}` would still trigger a re-render. So, I think it makes sense to use `useCallback` here, as well as `useMemo`. @hannojg, what do you think?

Reply:
Not sure if i agree with re-render for button - if it will be correctly specified with React.memo - it will be re-renders only if any of the props will be changed. Also if we talking about FC - i see that a lot of components still not have memo wrappers - which means that most of the useCallbacks still not needed, as an example in your code a little above src/components/EmojiPicker/EmojiPickerButtonDropdown.js - you have onPress function which is not wrapped in useCallback, because PressableWithoutFeedback - is not memoized. Also your useCallback will be re-creates every time because defaultEmoji, and defaultText are not memoized and will be new for any re-render. And making them memoized doesn't make any sense because both are not expensive calculations - as the result - memoized all that items will not bring and performance improvements but will keep more memory :-)